driver/TrEE/TrEE/OpteeFtpmService.h

```diff
@@ -0,0 +1,221 @@
+
+enum {
+    TpmStateReady = 0,
+    TpmStateIdle
+};
+
+#define CONTROL_AREA_REQUEST_CMD_READY  (0x00000001)
+#define CONTROL_AREA_REQUEST_GOIDLE     (0x00000002)
+
+#define CONTROL_AREA_STATUS_ERROR       (0x00000001)
+#define CONTROL_AREA_STATUS_IDLE        (0x00000002)
+
+#define CONTROL_AREA_START_START        (0x00000001)
+
+typedef struct _FTPM_CONTROL_AREA
+{
+    //
+    // This used to a Reserved field. This is the Miscellaneous field for the Command/Response interface.
+    // 
+    volatile UINT32  Request;
+    
+    //
+    // The Status field of the Control area.
+    //
+    volatile UINT32  Status;
+    
+    //
+    // The Cancel field of the Control area. TPM does not modify this field, hence it is not declared volatile.
+    //
+    UINT32  Cancel;
+    
+    //
+    // The Start field of the Control area.
+    //
+    volatile UINT32  Start;
+
+    //
+    // The control area is in device memory. Device memory
+    // often only supports word sized access. Split all
+    // 64bit fields into High and Low parts.
+    //
+    UINT32 InterruptControlLow;
+    UINT32 InterruptControlHigh;
+
+    //
+    // Command buffer size.
+    //
+    UINT32 CommandBufferSize;
+
+    //
+    // The control area is in device memory. Device memory
+    // often only supports word sized access. Split all
+    // 64bit fields into High and Low parts.
+    //
+    // Command buffer physical address.
+    //
+    UINT32 CommandPALow;
+    UINT32 CommandPAHigh;
+
+    //
+    // Response Buffer size.
+    //
+    UINT32 ResponseBufferSize;
+
+    //
+    // The control area is in device memory. Device memory
+    // often only supports word sized access. Split all
+    // 64bit fields into High and Low parts.
+    //
+    // Response buffer physical address.
+    //
+    UINT32 ResponsePALow;
+    UINT32 ResponsePAHigh;
+} FTPM_CONTROL_AREA, *PFTPM_CONTROL_AREA;
+
+//
+// From UEFI Tcg2PhysicalPresenceData.h
+//
+
+//
+// Non-volatile varaibles used to pass PPI data to UEFI
+//
+GUID PpiVariableGuid = {
+	0xaeb9c5c1, 0x94f1, 0x4d02, { 0xbf, 0xd9, 0x46, 0x2, 0xdb, 0x2d, 0x3c, 0x54 }
+};
+
+#define PPI_VARIABLE_NAME L"Tcg2PhysicalPresence"
+UNICODE_STRING PpiVariableNameUnicode = {
+		40,
+		42,
+		PPI_VARIABLE_NAME
+};
+
+typedef struct _TCG2_PPI_VARIABLE {
+	UINT8   Request;
+	UINT32  RequestParameter;
+	UINT8   LastRequest;
+	UINT32  Response;
+} TCG2_PPI_VARIABLE, *PTCG2_PPI_VARIABLE;
+
+#define PPI_FLAGS_VARIABLE_NAME L"Tcg2PhysicalPresenceFlags"
+UNICODE_STRING PpiFlagsVariableNameUnicode = {
+		50,
+		52,
+		PPI_FLAGS_VARIABLE_NAME
+};
+
+//
+// From UEFI Tcg2PhysicalPresenceLib.h
+//
+
+//
+// UEFI TCG2 library definition bit of the BIOS TPM Management Flags
+//
+#define TCG2_BIOS_TPM_MANAGEMENT_FLAG_PP_REQUIRED_FOR_CLEAR 0x02
+
+//
+// From UEFI TcgPhysicalPresence.h
+//
+
+//
+// TCG PP definition for physical presence ACPI function
+//
+#define TCG_ACPI_FUNCTION_GET_PHYSICAL_PRESENCE_INTERFACE_VERSION		1

```

Comment:
Indentation is off for all the constants.

Reply:
VS added tabs to some lines, fixed.